Parser/src/main/java/org/openzen/zenscript/parser/expression/ParsedExpressionArray.java

```diff
@@ -82,9 +82,18 @@ public CastedExpression cast(CastedEval cast) {
 			if (!maybeArray.isPresent()) {
 				// try implicit constructor instead
 				ResolvedType resolvedType = compiler.resolve(cast.type);
-				return resolvedType.findImplicitConstructor()
-						.map(constructor -> constructor.casted(compiler, position, cast, null, this))
-						.orElseGet(() -> cast.invalid(CompileErrors.invalidArrayType(cast.type)));
+				Optional<CastedExpression> implicitConstructor = resolvedType.findImplicitConstructor()
```

Comment:
In theory this and check and the `cast.of(eval())` branch cause implicit ctors to be checked twice.
But if we remove this here, some matching didn't work so I left it in